src/covvfit/plotting/_grid.py

```diff
@@ -58,3 +60,119 @@ def plot_grid(
+    """A two-dimensional grid of axes.
+
+    Attrs:
+        fig: Matplotlib figure.
+        axes: one-dimensional array of active axes,
+            with length equal to the number of active plots
+        axes_grid: two-dimensional array of all axes.
+
+
+    Note:
+        The number of plots in `axes_grid` is typically
+        greater than the one in `axes`, as `axes_grid`
+        contains also the axes which are not active
+    """
+
+    fig: Figure
+    axes: np.ndarray
+    axes_grid: np.ndarray
+
+    @property
+    def n_active(self) -> int:
+        return len(self.axes)
+
+    def map(
+        self,
+        func: Callable[[Axes], None] | Callable[[Axes, Any], None],
+        arguments: list | None = None,
+    ) -> None:
+        """Applies a function to each active plotting axis.
+
+        Args:
+            func: function to be applied. It can have
+                signature func(ax: plt.Axes)
+                if `arguments` is None, which modifies
+                the axis in-place.
+
+                If `arguments` is not None, then the function
+                should have the signature
+                func(ax: plt.Axes, argument)
+                where `argument` is taken from the `arguments`
+                list
+        """
+        if arguments is None:
+            for ax in self.axes:
+                func(ax)
+        else:
+            if self.n_active != len(arguments):
+                raise ValueError(
+                    f"Provide one argument for each active axis, in total {self.n_active}"
+                )
+            for ax, arg in zip(self.axes, arguments):
+                func(ax, arg)
+
+    def set_titles(self, titles: list[str]) -> None:
+        for title, ax in zip(titles, self.axes):
+            ax.set_title(title)
+
+    def set_xlabels(self, labels: list[str]) -> None:
+        for label, ax in zip(labels, self.axes):
+            ax.set_xlabel(label)
+
+    def set_ylabels(self, labels: list[str]) -> None:
+        for label, ax in zip(labels, self.axes):
+            ax.set_ylabel(label)
+
+
+def _calculate_nrows(n: int, ncols: int):
+    if ncols < 1:
+        raise ValueError(f"ncols has to be at least 1, was {ncols}.")
+    return int(np.ceil(n / ncols))
+
+
+def arrange_into_grid(
+    nplots: int,
+    ncols: int = 2,
+    axsize: tuple[float, float] = (2.0, 1.0),
+    **kwargs,
+) -> ArrangedGrid:
+    """Builds an array of plots to accommodate
+    the axes listed.
+
+    Args:
+        nplots: number of plots
+        ncols: number of columns
+        axsize: axis size
+        kwargs: keyword arguments to be passed to
+            `subplots_from_axsize`. For example,
+            ```
+            wspace=0.2,  # Changes the horizontal spacing
+            hspace=0.3,  # Changes the vertical spacing
+            left=0.5,    # Changes the left margin
+            ```
+    """
+    nrows = _calculate_nrows(nplots, ncols=ncols)
+
+    fig, axs = subplots_from_axsize(
```

Comment:
This function comes from a [handy small package](https://pypi.org/project/subplots-from-axsize/) created by Frederic. 

I added it as a dependency to `pyproject.toml`, so you may need to install it with Poetry (`poetry update`) or PIP (`pip install subplots-from-axsize`).

Reply:
Thanks Frederic :) super useful